sktime/transformations/compose.py

```diff
@@ -0,0 +1,502 @@
+        if names are non-unique, `f"_{str(i)}"` is appended to each name string
+            where `i` is the total count of occurrence of a non-unique string
+            inside the list of names leading up to it (inclusive)
+
+    `TransformerPipeline` can also be created by using the magic multiplication
+        on any transformer, i.e., any estimator inheriting from `BaseTransformer`
+            for instance, `my_trafo1 * my_trafo2 * my_trafo3`
+            will result in the same object as  obtained from the constructor
+            `TransformerPipeline([my_trafo1, my_trafo2, my_trafo3])`
+        magic multiplication can also be used with (str, transformer) pairs,
+            as long as one element in the chain is a transformer
+
+    Parameters
+    ----------
+    transformers : list of sktime transformers, or
+        list of tuples (str, transformer) of sktime transformers
+        these are "blueprint" transformers, states do not change when `fit` is called
+
+    Attributes
+    ----------
+    transformers_ : list of tuples (str, transformer) of sktime transformers
+        clones of transformers in `transformers` which are fitted in the pipeline
+        is always in (str, transformer) format, even if transformers is just a list
+        strings not passed in transformers are unique generated strings
+        i-th transformer in `transformers_` is clone of i-th in `transformers`
+    """
+
+    _required_parameters = ["transformers"]
+
+    # no default tag values - these are set dynamically below
+
+    def __init__(self, transformers):
+
+        self.transformers = transformers
+        self.transformers_ = self._check_estimators(transformers)
+
+        super(TransformerPipeline, self).__init__()
+
+        first_trafo = self.transformers_[0][1]
+        last_trafo = self.transformers_[-1][1]
+
+        self.clone_tags(first_trafo, ["X_inner_mtype", "scitype:transform-input"])
+        self.clone_tags(last_trafo, "scitype:transform-output")
+
+        self._anytag_notnone_set("y_inner_mtype")
+        self._anytag_notnone_set("scitype:transform-labels")
+
+        self._anytagis_then_set("scitype:instancewise", False, True)
+        self._anytagis_then_set("X-y-must-have-same-index", True, False)
+        self._anytagis_then_set("fit-in-transform", False, True)
+        self._anytagis_then_set("transform-returns-same-time-index", False, True)
+        self._anytagis_then_set("skip-inverse-transform", True, False)
+        self._anytagis_then_set("capability:inverse_transform", False, True)
+        self._anytagis_then_set("handles-missing-data", False, True)
+        self._anytagis_then_set("univariate-only", True, False)
+
+    @property
+    def _transformers(self):
+        return self._get_estimator_tuples(self.transformers, clone_ests=False)
+
+    @_transformers.setter
+    def _transformers(self, value):
+        self.transformers = value
+
+    def __mul__(self, other):
```

Comment:
I would remove the two dunder methods

Reply:
why? If one does not like them, one can simply not use them.
I think it does add a lot of user comfort.